tests-e2e/core/dt_dtc_dtclass_test.go

```diff
@@ -0,0 +1,421 @@
+			dtc, matchingDT, expectedSpaceRequest := createTest(appstudiosharedv1.ReclaimPolicy_Delete)
+
+			By("Deletion step 1 - Delete the DeploymentTargetClaim")
+			err := k8sClient.Delete(context.Background(), &dtc)
+			Expect(err).To(BeNil())
+
+			By("Deletion step 2 - Ensure the matching DT is also starting to delete")
+			Eventually(&matchingDT, "60s", "1s").Should(k8s.HasNonNilDeletionTimestamp(k8sClient))
+
+			By("Deletion step 3 - Ensure the matching SpaceRequest is also starting to delete")
+			Eventually(&expectedSpaceRequest, "60s", "1s").Should(k8s.HasNonNilDeletionTimestamp(k8sClient))
+
+			By("Deletion step 4 - Update the fake Space Request to Terminating")
+
+			err = spacerequest.UpdateStatusWithFunction(&expectedSpaceRequest, func(spaceRequestParam *codereadytoolchainv1alpha1.SpaceRequestStatus) {
+
+				spaceRequestParam.Conditions = []codereadytoolchainv1alpha1.Condition{
+					{
+						Type:   codereadytoolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionTrue,
+						Reason: codereadytoolchainv1alpha1.SpaceTerminatingReason,
+					},
+				}
+			})
+			Expect(err).To(BeNil())
+
+			Consistently(&expectedSpaceRequest, "10s", "1s").Should(k8s.ExistByName(k8sClient),
+				"the space request should not be deleted while it is terminating")
+			Consistently(&matchingDT, "10s", "1s").Should(k8s.ExistByName(k8sClient), "the DT should not be deleted while it is terminating")
+
+			err = spacerequest.UpdateStatusWithFunction(&expectedSpaceRequest, func(spaceRequestParam *codereadytoolchainv1alpha1.SpaceRequestStatus) {
+
+				spaceRequestParam.Conditions = []codereadytoolchainv1alpha1.Condition{
+					{
+						Type:   codereadytoolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionTrue,
+						Reason: codereadytoolchainv1alpha1.SpaceTerminatingFailedReason,
+					},
+				}
+			})
+			Expect(err).To(BeNil())
+
+			Eventually(matchingDT, "3m", "1s").Should(dtfixture.HasStatusPhase(appstudiosharedv1.DeploymentTargetPhase_Failed))
+
+			By("We now simulate the SpaceRequest finally cleaning itself up, which should cause the DT and DTC to be deleted")
+
+			By("removing the finalizer from the SpaceRequest, which should cause it to be deleted.")
+			expectedSpaceRequest.Finalizers = []string{}
+			err = k8s.Update(&expectedSpaceRequest, k8sClient)
+			Expect(err).To(BeNil())
+
+			Eventually(&expectedSpaceRequest, "10s", "1s").ShouldNot(k8s.ExistByName(k8sClient))
+
+			By("deletion step 5 - no-op")
+
+			By("deletion step 6 - the DT and DTC should also be deleted, once the SpaceRequest is removed")
+
+			Eventually(&matchingDT, "30s", "1s").ShouldNot(k8s.ExistByName(k8sClient))
+			Eventually(&dtc, "30s", "1s").ShouldNot(k8s.ExistByName(k8sClient))
+
+		})
+
+		It("should ensure that if the SpaceRequest fails to delete, DT is set to failed, and that if the SpaceRequest is eventually deleted, so will the DT and DTC", func() {
```

Comment:
I think we should update the description of this test case. We are deleting the DT and verifying if the DTC still exists.

Reply:
@chetan-rns whoops, great catch!